openstack-openedx/vars-template.yml

```diff
@@ -0,0 +1,400 @@
+################################################################################
+### Core settings
+
+# Set this to the domain name of your site
+COMMON_HOSTNAME: 'www.example.com'
+# This should be a slug (no spaces or special characters) representing your organization name
+COMMON_ENVIRONMENT: myorg
+# This should be a slug (no spaces or special characters) identifying this site
+COMMON_DEPLOYMENT: mysite
+
+# The name of your site
+EDXAPP_PLATFORM_NAME: "SET ME to site name"
+EDXAPP_SITE_NAME: '{{ COMMON_HOSTNAME }}'
+EDXAPP_LMS_NGINX_PORT: 80
+EDXAPP_LMS_SSL_NGINX_PORT: 443
+# Set this to 'https' or 'http' depending on which one you're using
+EDXAPP_LMS_BASE_SCHEME: 'https'
+EDXAPP_LMS_SITE_NAME: '{{ EDXAPP_SITE_NAME }}'
+EDXAPP_LMS_BASE: '{{ EDXAPP_SITE_NAME }}'
+
+# Set this to a unique domain name used by staff to preview changes to courseware content.
+# This domain must be different from EDXAPP_LMS_SITE_NAME
+EDXAPP_PREVIEW_LMS_BASE: 'preview.example.com'
+EDXAPP_LMS_PREVIEW_NGINX_PORT: 80
+
+# Set this to the domain name you want to use for Studio:
+EDXAPP_CMS_SITE_NAME: 'studio.example.com'
+EDXAPP_CMS_BASE: '{{ EDXAPP_CMS_SITE_NAME }}'
+CMS_HOSTNAME: '{{ EDXAPP_CMS_SITE_NAME }}'
+EDXAPP_CMS_NGINX_PORT: 80
+EDXAPP_CMS_SSL_NGINX_PORT: 443
+
+# Set this to a string for the part of your domain, such as ".example.com"
+# that is shared by the LMS and Studio, to enable session sharing between
+# LMS and the Studio.
+#EDXAPP_SESSION_COOKIE_DOMAIN: '.example.com'
+
+################################################################################
+### HTTPS, proxy, and redirect settings
+
+# Set to false if deployed behind a proxy or load balancer.
+NGINX_SET_X_FORWARDED_HEADERS: True
```

Comment:
This is a little confusing..  Above, we're recommending `EDXAPP_LMS_BASE_SCHEME: 'https'` by default, but also setting `NGINX_ENABLE_SSL: false` below, which makes it look like we're recommending a load balancer, with SSL termination.  So then `NGINX_SET_X_FORWARDED_HEADERS` should be `False`.

If on the other hand, this is for a single server, https deployment, then we should make that clearer, by uncommenting `NGINX_SSL_CERTIFICATE` and `NGINX_SSL_KEY` below, with `"SET ME"` as values like the other required variables.

I'm not sure which situation is better to recommend, but we should pick one and be consistent.